src/service/poc_mobile.proto

```diff
@@ -38,4 +38,132 @@ service poc_mobile {
   rpc submit_speedtest(speedtest_req_v1) returns (speedtest_resp_v1);
   rpc submit_cell_heartbeat(cell_heartbeat_req_v1)
       returns (cell_heartbeat_resp_v1);
-}
\ No newline at end of file
+}
+
+message file_info {
+  string key = 1;
+  file_type file_type = 2;
+  uint64 timestamp = 3;
+  uint64 size = 4;
+}
+
+message processed_files { repeated file_info files = 1; }
+
+message share {
+  string cbsd_id = 1;
+  bytes pub_key = 2;
+  uint64 weight = 3;
+  uint64 timestamp = 4;
+  cell_type cell_type = 5;
+}
+
+message shares { repeated share shares = 1; }
+
+message invalid_share {
+  string cbsd_id = 1;
+  bytes pub_key = 2;
+  uint64 weight = 3;
+  uint64 timestamp = 4;
+  cell_type cell_type = 5;
+  string invalid_reason = 6;
+}
+
+message invalid_shares { repeated invalid_share shares = 1; }
+
+message speed_share {
+  bytes pub_key = 1;
+  /// bytes/sec
+  uint64 upload_speed = 2;
+  /// bytes/sec
+  uint64 download_speed = 3;
+  uint32 latency = 4;
+  uint64 timestamp = 5;
+}
+
+message speed_share_list {
+  bytes gw_pub_key = 1;
+  repeated speed_share speed_shares = 2;
+}
+
+message speed_shares { repeated speed_share_list speed_shares = 1; }
+
+message invalid_speed_share {
+  bytes pub_key = 1;
+  /// bytes/sec
+  uint64 upload_speed = 2;
+  /// bytes/sec
+  uint64 download_speed = 3;
+  uint32 latency = 4;
+  uint64 timestamp = 5;
+  string invalid_reason = 6;
```

Comment:
do we need a string here or is this something we can reasonably enumerate like an http error?

Reply:
I think we can enumerate it but I'm not sure that I can by myself. The only invalid reason I found was "not in time range"

Reply:
> I think we can enumerate it but I'm not sure that I can by myself. The only invalid reason I found was "not in time range"

Well, there's definitely multiple invalid reasons, some being:
- heartbeat decode failed for some reason, although ingester service might not allow that cc: @madninja 
- gateway owner not found
- heartbeat not within time range
- unsupported cbsd_id -> cell_type